packages/store/src/storage.ts

```diff
@@ -1345,7 +1345,8 @@ class Storage {
       // if uploaded tarball has a different shasum, it's very likely that we
       // have some kind of error
       if (validatioUtils.isObject(metadata.dist) && _.isString(metadata.dist.tarball)) {
-        const tarball = metadata.dist.tarball.replace(/.*\//, '');
+        // anchor regex to avoid polynominal runtime
+        const tarball = metadata.dist.tarball.replace(/^.*\//, '');
```

Comment:
I think would be nice if we move this out of an util function to be able to test it easily

Reply:
While looking for a place to put the function, I found that one already exists :-)